Addition of external_app

frkv opened the pull request.

frkv committed:
NCP-1945: Adding external app parameter and usage
NCP-1945: Adding new generated dfu_cc_pb2.py with addition of new FwType: EXTERNAL_APPLICATION
NCP-NONE: Moving OpCode Init to top-level

frkv committed:
NCP-1904: Adding default=False for --external-app and making it into a flag type variable
NCP-NONE: Fixing bool expression

stianrh committed:
add boot vaidation

stianrh committed:
protobuf files for boot validation

frkv committed:
NRF-1904: Allowing --sd-req to not be set for --external_app
NRF-1904: Fixing boolean error for --external_app

frkv committed:
Merge remote-tracking branch 'firstguy/feature/zigbee_dfu_image'

* firstguy/feature/zigbee_dfu_image:
  Add Zigbee DFU image support

# Conflicts:
#	nordicsemi/__main__.py
#	nordicsemi/dfu/package.py

frkv committed:
NRFFOSDK-12436: Undoing mergetool mishap
NRFFOSDK-12436: Making Zigbee packages external-app only

frkv committed:
NRFFOSDK-12346: Adding restriction for either --application, --softdevice or --bootloader for Zigbee FW upgrade

frkv committed:
NRFFOSDK-12346: Yet more mergetool auto-merge issues resolved

oyvindronningstad commented on nordicsemi/dfu/package.py:
```diff
@@ -371,7 +382,7 @@ def generate_package(self, filename, preserve_work_dir=False):
             bl_size = 0
             app_size = 0
-            if key == HexType.APPLICATION:
+            if key == HexType.APPLICATION or key == HexType.EXTERNAL_APPLICATION:
```
Could be:

    if key in [HexType.APPLICATION, HexType.EXTERNAL_APPLICATION]

frkv replied:
Changing it

oyvindronningstad commented on nordicsemi/dfu/package.py:
```diff
@@ -156,10 +159,17 @@ def __init__(self,
-                                     firmware_version=app_version,
-                                     filename=app_fw,
-                                     init_packet_data=init_packet_vars)
+            if is_external:
```
Could be:

            firmware_type = HexType.EXTERNAL_APPLICATION if is_external else HexType.APPLICATION
            self.__add_firmware_info(firmware_type=firmware_type,
                                     firmware_version=app_version,
                                     filename=app_fw,
                                     init_packet_data=init_packet_vars)

frkv replied:
Changing it

frkv committed:
NRFFOSDK-12346: Fixing comments from pull request

frkv committed:
NRFFOSDK-12346: Fixing boolean test of None

frkv committed:
NRFFOSDK-12346: Adding --zigbee-ota-hw-version (required) and --zigbee-ota-fw-version (optional)

frkv committed:
NRFFOSDK-12346: Fixing issue with not opening OTA file as binary (too small images read, 260KB => 640 bytes)
NRFFOSDK-12346: Cleanup of the OTA_file class generation

oyvindronningstad committed:
Change CLI name from VALIDATE_SHA256 to VALIDATE_GENERATED_SHA256 because that is the implementation in the bootloader

oyvindronningstad committed:
Add documentation and error messages for arguments to --boot-validation

bihanssen committed:
Merge pull request #161 from tomchy/feature/zigbee_dfu_image

Add Zigbee DFU image support

stianrh committed:
Merge pull request #1 from oyvindronningstad/master-stianrh

Improve CLI of --boot-validation

Commit:
Refactored DFU Settings generation to allow for easier handling of non uint32 values

Commit:
Added version 2 for BL DFU Settings page

Commit:
Fixed minor bug in how crc was calculated from IntelHex object

frkv committed:
NRFFOSDK-NONE: adding comma in parameter list

Commit:
Merge remote-tracking branch 'wayland/master' into settings-page-boot-validation

frkv committed:
Merge branch 'master' of https://github.com/NordicSemiconductor/pc-nrfutil

* 'master' of https://github.com/NordicSemiconductor/pc-nrfutil:
  Add a warning about overwriting nRF DFU settings backup page
  Extend unit tests by backup page checks
  Add setting for DFU settings backup page placement
  Update DFU settings backup page together with DFU settings
  BUGFIX: open binary file in a binary mode
  Remove additional commas inside README.md

# Conflicts:
#	nordicsemi/dfu/package.py

frkv committed:
protobuf files for boot validation

# Conflicts:
#	nordicsemi/dfu/dfu-cc.options
#	nordicsemi/dfu/dfu-cc.proto
#	nordicsemi/dfu/dfu_cc_pb2.py

frkv committed:
NRFFOSDK-NONE: Removing pycharm debug code

frkv committed:
Merge branch 'settings-page-boot-validation' of https://github.com/andvib/pc-nrfutil

* 'settings-page-boot-validation' of https://github.com/andvib/pc-nrfutil:
  Fixed minor bug in how crc was calculated from IntelHex object
  Added version 2 for BL DFU Settings page
  Refactored DFU Settings generation to allow for easier handling of non uint32 values
  Add documentation and error messages for arguments to --boot-validation
  Change CLI name from VALIDATE_SHA256 to VALIDATE_GENERATED_SHA256 because that is the implementation in the bootloader
  protobuf files for boot validation
  add boot vaidation

# Conflicts:
#	nordicsemi/__main__.py

frkv committed:
NRFFOSDK-NONE: Adding boot validation to NONE for the outter zigbee packet

frkv commented:
Closing as this is old
